src/ExEss/Cms/Http/EventSubscriber/LogSubscriber.php

```diff
@@ -0,0 +1,145 @@
+    {
+        $this->logPsrRequest($this->getPsrRequest($event->getRequest()));
+    }
+
+    public function logResponse(ResponseEvent $event): void
+    {
+        $this->logPsrResponse(
+            $this->getPsrRequest($event->getRequest()),
+            $this->getPsrResponse($event->getResponse())
+        );
+    }
+
+    public function logPsrRequest(ServerRequestInterface $request): void
+    {
+        $mode = $this->determineLogMode($request);
+
+        $request->getBody()->rewind();
+
+        $message = 'Incoming call for: ' . $request->getUri()->getPath();
+        $message .= ', method: ' . $request->getMethod();
+        if ($mode === self::LOG_BODY) {
+            $message .= ', body: ' . $this->getTruncated($request->getBody());
+        } elseif ($mode === self::LOG_MODEL) {
+            $requestModel =  DataCleaner::getCleanedModel($request->getParsedBody()['model'] ?? []);
+            $action = ((array) $request->getParsedBody())['action'] ?? [];
+            $requestEvent = ((array) $action)['event'] ?? FlowAction::EVENT_INIT;
+            $message .= ', event: ' . \json_encode($requestEvent) . ', model: ' . \json_encode($requestModel);
+        }
+
+        $this->logger->info($message);
+    }
+
+    public function logPsrResponse(ServerRequestInterface $request, ResponseInterface $response): void
+    {
+        $mode = $this->determineLogMode($request);
+
+        $response->getBody()->rewind();
+
+        $message = 'Outgoing response for: ' . $request->getUri()->getPath()
+            . ', method: ' . $request->getMethod()
+            . ', status: ' . $response->getStatusCode()
+            . ', message: ' . $response->getReasonPhrase();
+        if ($mode === self::LOG_BODY) {
+            $message .= ', body: ' . $this->getTruncated($response->getBody());
+        } elseif ($mode === self::LOG_MODEL) {
+            $responseModel = DataCleaner::getCleanedModel(
+                \json_decode($response->getBody()->getContents(), true)['data']['model'] ?? []
+            );
+            $message .= ', model: ' . \json_encode($responseModel);
+        }
+
+        $this->logger->info($message);
+    }
+
+    private function getTruncated(StreamInterface $stream): string
+    {
+        return $stream->getSize() > self::MAX_MESSAGE_LENGTH ?
+            $stream->read(self::MAX_MESSAGE_LENGTH) .  self::SUFFIX_TRUNCATED :
+            (string) $stream
+        ;
+    }
+
+    private function determineLogMode(ServerRequestInterface $request): string
+    {
+        if ($request->getMethod() === Request::METHOD_GET) {
+            return self::LOG_NO_CONTENT;
+        }
+
+        // @todo check for route name instead of uri path
+        // locally it starts with /, on AWS it doesn't
+        $pos = \strpos($request->getUri()->getPath(), '/Api/Flow');
+        if ($pos === 0 || $pos === 1) {
+            return self::LOG_MODEL;
+        }
+
+        if (!\preg_match('~/login$~', $request->getUri()->getPath())) {
```

Comment:
we could avoid the preg_match here by checking the full uri, but maybe hasn't too much an impact since it's only twice per request

Reply:
Well I added an @todo just above it for that reason, but need them all routed through Symfony first (for login I could do it already but prefer all together)

The path comparison is pointless in SF